### ISSUE TITLE
app-tap/http_client.test.lua flaky fails on Mac OS

### ISSUE THREAD

**Issue by Totktonada** [tarantool/tarantool#4288]
Tarantool version: 2.2.0-418-gf824d0927 Darwin-x86_64-Debug.
OS version: Mac OS X Sierra 10.12.6 16G29 (tt-mac).

brew now contains curl-7.65.1 which affected by https://github.com/curl/curl/issues/3995 (this problem leads to segfaults). The next version is not released yet. Linux distros are not affected AFAIK, because don't update curl so fast (at least Gentoo).

The issue hits us in [Travis-CI][1]. It can be reproduced w/o test-run in the following way:

```sh
$ ./src/tarantool -e "for i = 1, 100 do print(require('json').encode(require('http.client').get('http://localhost:1').status)) end os.exit()"
```

I made the following steps on tt-mac:

* brew edit curl; reverted https://github.com/Homebrew/homebrew-core/pull/40717/files
* brew reinstall curl

After that segfaults are gone.

We possibly should find a way to do so in CI (non-iteractively, it seems we can patch /usr/local/Homebrew/Library/Taps/homebrew/homebrew-core/Formula/curl.rb and run `brew reinstall curl`) or just ignore and wait for the next curl release.

[1]: https://travis-ci.org/tarantool/tarantool/jobs/545711320#L4723

**Comment by Totktonada** [comment]
> [curl] 7.65.2 is expected to ship on July 17, 2019.

https://github.com/curl/curl/issues/4057#issuecomment-504230933

We need to resolve it on our side. @avtikhon, can you take a look?

**Comment by Totktonada** [comment]
The simple way to install curl-7.65.0:

```sh
brew install https://raw.githubusercontent.com/Homebrew/homebrew-core/e47f3e2682fa3d8a6ee58ded40718a1326e08a6d/Formula/curl-openssl.rb
```

https://github.com/Homebrew/homebrew-core/issues/40812#issuecomment-500197510

**Comment by Totktonada** [comment]
The idea from @Mons: don't run at all if such curl version is detected.

**Comment by Totktonada** [comment]
7.65.2 was released (and 7.65.3 two days after that):

https://curl.haxx.se/changes.html#7_65_2
https://github.com/Homebrew/homebrew-core/pull/42065

Let's remove the workaround.

**Comment by Totktonada** [comment]
The workaround was reverted in abd5ccee55b48526f45ec0088ccc70fbdcad90b1 (master), de56395b7902a16bbe67b54cc8aaa7fe30d840a3 (2.1) and ef7ed7b099b1b2b86877293e119c8b9cf450b6b1 (1.10).
